scaladoc/src/dotty/tools/scaladoc/tasty/Scaladoc2AnchorCreator.scala

```diff
@@ -7,10 +7,67 @@ import scala.util.matching.Regex
 
 object Scaladoc2AnchorCreator:
 
-  def getScaladoc2Type(using Quotes)(t: reflect.Tree) =
-    import reflect.*
-    val regex = t match
-      case d: DefDef =>  "def"
-      case t: TypeDef => "type"
-      case v: ValDef =>  "val|var"
-    t.show(using Printer.TreeShortCode).split(regex, 2)(1).replace(" ","")
+  def getScaladoc2Type(using Quotes)(sym: quotes.reflect.Symbol) = signatureAnchor(sym)
+
+  /** Creates the signature anchor
+   *
+   *  - `X` for a `type X ...`
+   *  - `x:X` for a `val x: X`
+   *  - `f[U1,...](x1:T1,...)...:R` for a `def f[U1, ...](x1: T1, ...)...: R`
+   *
+   *  Types are printed without their paths. No spaces are printed in the output
+   *  except for the one after `(implicit`.
+   */
+  private def signatureAnchor(using Quotes)(sym: quotes.reflect.Symbol): String =
+    import quotes.reflect.*
+    def signatureType(tp: quotes.reflect.TypeRepr): String =
+      tp match
+        case mt @ MethodType(paramNames, paramTypes, res) =>
+          val openClause = if mt.isImplicit then "(implicit " else "("
```

Comment:
There are two things that need changing here:
1. There should be no space after `implicit` i.e. in the anchor `implicit` should be merged with the parameter name.
2. Every parameter in an implicit parameter list should have a prepended `implicit` to it e.g. https://www.scala-lang.org/api/2.13.10/scala/collection/ArrayOps.html#flatten[B](implicitasIterable:A=%3Escala.collection.IterableOnce[B],implicitm:scala.reflect.ClassTag[B]):Array[B]